FXyz-Samples/src/main/java/org/fxyz3d/samples/utilities/RayShooting.java

```diff
@@ -74,33 +77,32 @@ public class RayShooting extends ShapeBaseSample {
     
     private final AmbientLight rayLight = new AmbientLight();
     protected PointLight light, light2, light3;
-    private boolean fireRay = true;
     protected Sphere target1, target2;
-
+    protected Box origin;
+    
     @Override
     protected void addMeshAndListeners() {
-              
-        
-        camera.setTranslateZ(-5000);
-        //First person shooter keyboard movement
-        subScene.setOnKeyPressed(ke -> {
-            // add a flag so we can still move the camera
-            if (ke.getCode().equals(KeyCode.CONTROL)) {
-                fireRay = false;
-            }
-            
-        });
-        
-        subScene.setOnKeyReleased(ke -> {
-            // release flag
-            if (ke.getCode().equals(KeyCode.CONTROL)) {
-                fireRay = true;
-            }
+        camera.setTranslateZ(-2000);
+
+        subScene.addEventHandler(KeyEvent.KEY_PRESSED, event -> {
+            if(event.isControlDown()) {
```

Comment:
Why do you need Ctrl here? Typically you could press WASD, Space, Z without the need of any modifier.
Ctrl+{WASD/Space/Z} might have system implications in different platforms.

Reply:
I wanted to allow the user to translate the origin cube (origin of the ray) around in space but the ShapeBaseExample already had wasd etc mapped to translating the camera.

Reply:
Okay, but as I mentioned, Ctrl+{...} has some undesired effects on Mac for instance.

Reply:
oh. crud. Ok I'll switch to other key codes without control. one minute...

Reply:
see latest.